File: codes/yujo/169_majorityelement.ts
function majorityElement(nums: number[]): number {
  const map: Record<string, number> = {};

  for (const num of nums) {
    if (map[num]) {
      map[num] = map[num] += 1;
    } else {
      map[num] = 1;
    }
  }

  const maxValue = Math.max(...Object.values(map));
  const maxKey = Object.keys(map).find((key) => map[key] === maxValue);

  return Number(maxKey);
}

File: codes/yunyeong/169_majorityelement.py
class Solution(object):
    def majorityElement(self, nums):
        flag = int(len(nums)/2)
        counter = {}
        for num in nums:
            if num in counter:
                counter[num] += 1
            else:
                counter[num] = 1
            if counter[num] > flag:
                return num
        return None
# O(n) / O(n) .